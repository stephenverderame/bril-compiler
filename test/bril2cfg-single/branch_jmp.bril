@main {
    foo: int = const 10;
    bar: int = const 20;
    b: bool = eq foo bar;
    br b .yes .no;
.yes:
    jmp .yes.0;
.no:
    jmp .yes;
.yes.0:
    jmp .yes.1;
.yes.1:
    print b;
}