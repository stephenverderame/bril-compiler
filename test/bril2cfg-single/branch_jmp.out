digraph {
	subgraph cluster_main {
		label="main()";
		rankdir="TB";
		main_0 [label="START", shape="rectangle", style="rounded"];
		main_2 [label="foo := 10", shape="rectangle", style="rounded"];
		main_3 [label="bar := 20", shape="rectangle", style="rounded"];
		main_4 [label="b := foo == bar", shape="rectangle", style="rounded"];
		main_5 [label="if b", shape="rectangle", style="rounded"];
		main_9 [label="print b", shape="rectangle", style="rounded"];

		main_0 -> main_2;
		main_2 -> main_3;
		main_3 -> main_4;
		main_4 -> main_5;
		main_5 -> main_9 [label="T"];
		main_5 -> main_9 [label="F"];
	}
}
